### ISSUE TITLE
Have color difference

### ISSUE THREAD

**Issue by kane9527-e** [kyubuns/AkyuiUnity#48]
1.Import  xd file :https://drive.google.com/file/d/1LAmtfFHxfgmo15xldfqP9qzH8GD26wmf/view?usp=sharing
2. Observe Prefab and find that some elements in the XD file have color difference
Version: main 
What is the reason for this? I don't quite understand. Please answer it, thank you!
<img width="1025" alt="f1e45408ddf18cd8b573c23bb9955cd" src="https://user-images.githubusercontent.com/9569527/218783721-479c7995-b727-4af4-a050-5c23ac0f1991.png">

**Comment by IRod22** [comment]
@kyubuns, @kane9527-e, it looks like AkyuiUnity is brightening the UI when it is converting the `.xd` file to Unity.

**Comment by kyubuns** [comment]
This problem is caused by AdobeXD not managing color profiles correctly.
https://adobexd.uservoice.com/forums/353007-adobe-xd-feature-requests/suggestions/17480899-color-management-set-color-profiles

I was left with the belief that AdobeXD would one day "normalize", but since I no longer expect AdobeXD to add new features
I am trying to figure out if this problem can be absorbed on the AkyuiUnity side, but the color profile issue is too complex to make any promises....

**Comment by IRod22** [comment]
@kyubuns, in all of the cases you have seen this bug, do all Akyui/Unity UI files look brighter than the original Xd files? If so, you could add a shading process while transforming the XD file.

**Comment by kyubuns** [comment]
I am not yet aware of the whole issue and will investigate it in due time.

**Comment by IRod22** [comment]
> I am not yet aware of the whole issue and will investigate it in due time.

When you do investigate this, create a private repository containing screenshots of UIs before and after being transformed by AkyuiUnity; once you take several before-and-after images, invite me as a collaborator, so I can compare and contrast the UIs and give feedback.

**Comment by IRod22** [comment]
I apologize if my [previous comment](https://github.com/kyubuns/AkyuiUnity/issues/48#issuecomment-1430677904) had a rude tone; I edited it, so it would be less rude and more clear.

**Comment by kyubuns** [comment]
> I apologize if my [previous comment](https://github.com/kyubuns/AkyuiUnity/issues/48#issuecomment-1430677904) had a rude tone; I edited it, so it would be less rude and more clear.

No, I appreciate your cooperation, I just don't know when I will be able to get started on this task.
And I don't want to leave unnecessary conversations on the issue.

**Comment by IRod22** [comment]
> And I don't want to leave unnecessary conversations on the issue.

I meant that the conversation can be on the repository itself.

**Comment by IRod22** [comment]
@kyubuns, @kane9527-e, I think I found a fix!
https://community.adobe.com/t5/adobe-xd-discussions/color-profile-problem/td-p/9946275

**Comment by kane9527-e** [comment]
Setting the SRGB as the monitor profile, this decision is bad. My team uses Windows and Mac together

**Comment by IRod22** [comment]
Never mind.

**Comment by IRod22** [comment]
@kyubuns, I could look into *possibly* installing Unity to help find a fix, but I'm not sure if my computer can handle a large game engine right now...

**Comment by IRod22** [comment]
> @kyubuns, I could look into _possibly_ installing Unity to help find a fix, but I'm not sure if my computer can handle a large game engine right now...

However, because I'm a web developer, I can use my knowledge of built-in image filters to see if a certain combination of filters will fix the issue. @kane9527-e, when you have a chance, do you mind taking a screenshot of the entire UI in Xd and a different screenshot of the prefab in Unity, please? That way I can get a fuller picture of what is going on.

**Comment by IRod22** [comment]
@kyubuns, @kane9527-e, I'm working on an [image filtering website](https://github.com/IRod22/what-happened-to-this-image) where two image URLs (the before and after pictures respectively) can be passed as inputs along with the required filter configurations. My website will also allow us to create permalinks, so we can share configuration settings. This will allow us to see whether AkyuiUnity will be able to "absorb the issue" or not; if so, then we know how to fix the bug.

**Comment by IRod22** [comment]
@kyubuns, @kane9527-e, y'all can disregard my previous comments. I was doing more research, and (so far) I found that Akyui will need to limit its color gamut to sRGB, which is an international standard used by web browsers, Unity, and possibly other game engines. This means that the `Xd -> Akyui` conversion process will need to dull the Adobe RGB color space to sRGB, and the `Akyui -> [game UI]` conversion process will need to make adjustments to the UI colors, so the UI is compatible with Unity and other game engines. In a nutshell, unless a UI format using the sRGB space is used as the input format, or there is a new and wider color spectrum standard, there is no way to avoid unexpected UI color issues.

#### References:
- https://expertphotography.com/srgb-vs-adobe-rgb/
- https://answers.unity.com/questions/1864025/can-i-use-larger-color-spaces-than-srgb-in-unity.html

**Comment by kane9527-e** [comment]
Thank you for your answers, but due to the color problem, I can only temporarily give up using this scheme in the project, but I still thank the author and your contributions

**Comment by IRod22** [comment]
@kane9527-e, you are welcome; I am sorry that a permanent fo could not be found...

**Comment by IRod22** [comment]
@kyubuns, do you want me to add a warning about this issue in Pyll Request #49?

**Comment by kyubuns** [comment]
Thanks for doing the research.
I don't plan to write in the documentation until I close this PR, as people who are having problems should come to the Issue first.
When I myself really give up, this PR will be closed.

**Comment by IRod22** [comment]
Understood.

**Comment by IRod22** [comment]
If you want me to conduct more research, don't hesitate to let me know.

**Comment by IRod22** [comment]
@kyubuns, I don't know if you're aware of this, but [Adobe is in the process of purchasing Figma](https://news.adobe.com/news/news-details/2022/Adobe-to-Acquire-Figma/default.aspx). I found some other articles saying that this may affect Adobe Xd, which may be discontinued after the agreement is finished. Some different articles say that Xd will be around a bit longer after this, but some different ones say that Xd is already lost; Adobe Xd's UserVoice forum shows posts of [users *wanting* Xd to be discontinued](https://adobexd.uservoice.com/forums/353007-adobe-xd-feature-requests/suggestions/45807016-discontinue-xd), and a totally different article I found is saying that [image-based UI design might be dead in the water entirely](https://techcrunch.com/sponsor/uxpin/image-based-design-is-dead-how-code-became-a-common-language-in-product-development/).

**Comment by kyubuns** [comment]
@IRod22 If the topic is not related to this issue, please do it elsewhere.

**Comment by IRod22** [comment]
Sorry! I'm deleting it now. 🤦‍♂️

**Comment by kane9527-e** [comment]
Whether to consider adding linear space and gamma space to judge whether the color of the generated image is correct automatically according to the project settings or the options provided

**Comment by kane9527-e** [comment]
Similar to shader code      
#if !defined(UNITY_COLORSPACE_GAMMA) && (UNITY_VERSION >= 550)
                    o.color.rgb = GammaToLinearSpace(v.color.rgb);
                    o.color.a = v.color.a;
#endif

**Comment by IRod22** [comment]
> Similar to shader code
> #if !defined(UNITY_COLORSPACE_GAMMA) && (UNITY_VERSION >= 550)
> o.color.rgb = GammaToLinearSpace(v.color.rgb);
> o.color.a = v.color.a;
> #endif

@kane9527-e I don't think Unity defines `UNITY_COLORSPACE_GAMMA` or `UNITY_VERSION` for the C# preprocessor. [See the Unity docs for more info](https://docs.unity3d.com/Manual/PlatformDependentCompilation.html).

**Comment by kane9527-e** [comment]
https://helpx.adobe.com/xd/help/color-management.html

**Comment by kyubuns** [comment]
What? That's great news!

**Comment by kane9527-e** [comment]
Yes ，But I haven't had time to test yet, I hope you can test it and write a prompt on doc.

**Comment by kane9527-e** [comment]
I test it  with XD 57 version ,svg have color difference ontinue to exist.

![Image](https://github.com/user-attachments/assets/d312f4f2-8647-46a8-b420-2b1afc492735)
